### PR TITLE
[2/X] Fix bugs and add sentiment analysis.

### PR DESCRIPTION
**Pull request by nikhitashanker**
-Move comment form content to new page to reduce number of network requests on index.html.
-Fix live server known issue by removing the use of imageService for Blobstore API
-Add Positive and negative tags to comments (Sentiment API)

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/BlobstoreServlet.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/blobstore-image")
+public class BlobstoreServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Serve the blob that corresponds to the key provided.
+    BlobKey blobKey = new BlobKey(request.getParameter("blob-key"));
```
Comment (liangqigong) [inline portfolio/src/main/java/com/google/sps/servlets/BlobstoreServlet.java:17]: Use a const for blob-key

File: portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java [hunk 2]
```diff
@@ -64,17 +65,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     // Get input from the form for comment text.
     String text = CommonUtils.getParameter(request, TEXT, /* DefaultValue= */ "");
 
+    // Get text with the sentiment tag added.
+    String textWithSentiment = null;
```
Comment (liangqigong) [inline portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:69]: Could you avoid null here?

http://www.javapractices.com/topic/TopicAction.do?Id=134

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 3]
```diff
@@ -2,19 +2,19 @@
...
 
-  public Comment(String commenterEmail, String commenterName, long id, String imageUrl, String text,
-      long timestamp) {
+  public Comment(String blobKeyString, String commenterEmail, String commenterName, long id,
+      String text, long timestamp) {
+    this.blobKeyString = blobKeyString;
     this.commenterEmail = commenterEmail;
     this.commenterName = commenterName;
     this.id = id;
-    this.imageUrl = imageUrl;
```
Comment (liangqigong) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:17]: Next time I think you can put removing image in a separate PR?

File: portfolio/src/main/java/com/google/sps/servlets/BlobstoreServlet.java (hunk 1 above)
Comment (nikhitashanker) [reply portfolio/src/main/java/com/google/sps/servlets/BlobstoreServlet.java:17]: Done.

File: portfolio/src/main/java/com/google/sps/data/Comment.java (hunk 3 above)
Comment (nikhitashanker) [reply portfolio/src/main/java/com/google/sps/data/Comment.java:17]: Ack - will do.

File: portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java (hunk 2 above)
Comment (nikhitashanker) [reply portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:69]: Done - I think it makes sense to set it to text.

File: portfolio/src/main/java/com/google/sps/utilities/SentimentUtils.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package com.google.sps.utilities;
+
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.Sentiment;
+import com.google.sps.data.Comment;
+import java.io.IOException;
+
+public class SentimentUtils {
+  public static final String POSITIVE_COMMENT_TAG = "POSITIVE COMMENT";
```
Comment (liangqigong) [inline portfolio/src/main/java/com/google/sps/utilities/SentimentUtils.java:10]: private here?

Comment (nikhitashanker) [reply portfolio/src/main/java/com/google/sps/utilities/SentimentUtils.java:10]: Done.
